src/community/oseo/oseo-core/src/main/java/org/geoserver/opensearch/eo/store/WorkspaceAwareDecoratingFeatureSource.java

```diff
@@ -0,0 +1,165 @@
+/* (c) 2023 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.opensearch.eo.store;
+
+import static org.geoserver.opensearch.eo.store.JDBCOpenSearchAccess.FF;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+import org.geoserver.catalog.WorkspaceInfo;
+import org.geoserver.security.decorators.DecoratingSimpleFeatureSource;
+import org.geotools.data.Query;
+import org.geotools.data.simple.SimpleFeatureCollection;
+import org.geotools.data.simple.SimpleFeatureSource;
+import org.geotools.factory.CommonFactoryFinder;
+import org.geotools.geometry.jts.ReferencedEnvelope;
+import org.opengis.filter.Filter;
+import org.opengis.filter.FilterFactory2;
+import org.opengis.filter.identity.FeatureId;
+
+/** Adds a workspace filter to the queries/filters, if a workspace is set */
+public class WorkspaceAwareDecoratingFeatureSource extends DecoratingSimpleFeatureSource {
```

Comment:
I'm all for descriptive names, but this is a bit too long. What about just ``WorkspaceFeatureSource``?